exceptions.py

```diff
@@ -20,3 +20,23 @@ class FilePermissionError(FileError):
 
 class DateError(Exception):
     pass
+
+
+class DataSourceError(Exception):
+    pass
+
+
+class UnsupportedDataSourceError(DataSourceError):
+    pass
+
+
+class DataSourceInterfaceError(DataSourceError):
```

Comment:
It's probably just `DataSourceError` or `InvalidImplementationError`, or something like that.